File: backend/entity/StudentInterest.go
package entity

import (
	"gorm.io/gorm"
)

type StudentInterest struct {
	gorm.Model
	InterestID uint     `json:"interest_id"`
	StudentID  uint     `json:"student_id"`
	Interest   Interest `gorm:"foreignKey:InterestID"`
	Student    Student  `gorm:"foreignKey:StudentID"`
}

File: frontend/src/interfaces/StudentInterest.ts
import type { InterestInterface } from './Interest';
import type { StudentInterface } from './Student';

export interface StudentInterestInterface {
  id?: number;  
  ID?: number;          // จาก gorm.Model.ID
  InterestID: number;
  StudentID: number;
  Interest?: InterestInterface;    // ความสัมพันธ์ (optional)
  Student?: StudentInterface;      // ความสัมพันธ์ (optional)
  CreatedAt?: string;
  UpdatedAt?: string;
  DeletedAt?: string | null;
}